### PR TITLE
Add dynamic/selectable theme support to the Advanced Compose add.

### PR DESCRIPTION
**Pull request by JohnNichol**
Give the user access to 4 different theme alternatives and a way to select them in the app

Screenshot_20220509_185158

Screenshot_20220509_185116
.

### CODE REVIEW

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/theme/ThemeScreen.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    ) {
+        item {
+            ListHeader {
+                Text("Color Schemes")
+            }
+        }
+        for (listItem in availableThemes) {
+            val checked = listItem.colors == currentlySelectedColors
+            item {
+                ToggleChip(
```
Comment (yschimke) [inline ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/theme/ThemeScreen.kt:54]: What about

```suggestion
MaterialTheme(colors = listItem.colors) {
                ToggleChip(
```

For a bit of pizzazz.

**Review by yschimke — APPROVED** [review]

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/WearApp.kt [hunk 2]
```diff
@@ -503,6 +512,29 @@ fun WearApp(
...
+                ) { it ->
+                    val scalingLazyListState = scalingLazyListState(it)
+                    val focusRequester = remember { FocusRequester() }
+
+                    ThemeScreen(
+                        scalingLazyListState = scalingLazyListState,
+                        focusRequester = focusRequester,
+                        currentlySelectedColors = themeColors,
+                        availableThemes = themeValues
+                    ) { colors -> themeColors = colors }
```
Comment (yschimke) [inline ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/WearApp.kt:534]: Curious: is there a agreed style for event handlers as last method param? named vs trailing lambda?

**Review by shumelchyk — APPROVED** [review]
Nice one 👍

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/theme/ThemeScreen.kt (hunk 1 above)
Comment (JohnNichol) [reply ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/ui/theme/ThemeScreen.kt:54]: As per offline discussion going to leave this as is. The override of selection control icon color is being done as the normal selection control color is the "secondary" color - so if you select the blue theme you get a green selection control color - which in this very specific use case is a potentially confusing given the colors are changing as you select items in the list.

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/WearApp.kt (hunk 2 above)
Comment (JohnNichol) [reply ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/WearApp.kt:534]: I think this falls under the general callback as last property to allow for lambdas unless it causes confusion. But I agree with compose where we will often have a content lambda at the end we will often end up naming callbacks. I think the basic premise should be considering have last param as something lambda'erable if you can.

**Pull request merged**